core/modules/system/system.module

```diff
@@ -2990,7 +2990,7 @@ function _system_rebuild_module_data() {
  *   Array of all available modules and their data.
  */
 function system_rebuild_module_data() {
-  $modules_cache = &backdrop_static(__FUNCTION__);
+  $modules_cache = &backdrop_static('modules_cache');
```

Comment:
The problem with this fix is that all calls to `backdrop_static_reset('system_rebuild_module_data')` become ineffective.

Reply:
Thanks for the update, @docwilmot, and yes, let's find the right solution. However, leaving the function as is also doesn't' seem right, because `$modules_cache` never gets populated, it is always empty.